admin/src/components/Generate/Generate.tsx

```diff
@@ -35,12 +39,17 @@ const Generate = ({ attributes, checkedAttributes, values, count, onChangeGenera
 		return faker.datatype.float({ min, max });
 	};
 
-	const getValueByStringAndRichtextType = (key: string): string => {
+	const getValueByStringAndRichtextType = (key: string, regex?: string): string => {
+		if (regex) {
+			return faker.helpers.fromRegExp(regex);
+		}
 		let { min, max } = values[key] as { min: number; max: number };
-		return faker.random.words(faker.datatype.number({ min, max }));
+		if (!min && !max) return faker.lorem.words();
+		if (min === 1 && max === 1) return capitalizeFirstLetter(faker.lorem.words(1));
+		return faker.lorem.words({ min, max });
 	};
 	const getValueByEmailType = (): string => {
-		return faker.internet.email();
+		return faker.internet.email({ provider: "element-software.co.uk" }).toLowerCase();
```

Comment:
As I understood from doc that provider will be randomly set if you don't set it
So for now you can just remove it(leave it as it was) and next mr you can add the ability to choose providers
Tnx!

Reply:
Hi yes that's right, I have removed it for this PR :)